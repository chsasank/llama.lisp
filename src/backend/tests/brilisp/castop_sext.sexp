;test for sext i1 true to i32 not completed
(brilisp
    (define ((print int) (n int)))

    (define ((main void))
        (set (a int8) (const -1))
        (set (f_n int) (sext a int))
        (set (tmp int) (call print f_n))

        (set (b int8) (const 127))
        (set (f_n int) (sext b int))
        (set (tmp int) (call print f_n))

        ; test zext too
        (set (f_n int) (zext a int))
        (set (tmp int) (call print f_n))
        (ret)))
