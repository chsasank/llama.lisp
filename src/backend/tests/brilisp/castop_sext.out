-1
127
255
